src/app-layer-modbus.c

```diff
@@ -37,1503 +37,21 @@
-#define MODBUS_FUNC_WRITESINGLECOIL     0x05
-#define MODBUS_FUNC_WRITESINGLEREG      0x06
-#define MODBUS_FUNC_READEXCSTATUS       0x07
-#define MODBUS_FUNC_DIAGNOSTIC          0x08
-#define MODBUS_FUNC_GETCOMEVTCOUNTER    0x0b
-#define MODBUS_FUNC_GETCOMEVTLOG        0x0c
-#define MODBUS_FUNC_WRITEMULTCOILS      0x0f
-#define MODBUS_FUNC_WRITEMULTREGS       0x10
-#define MODBUS_FUNC_REPORTSERVERID      0x11
-#define MODBUS_FUNC_READFILERECORD      0x14
-#define MODBUS_FUNC_WRITEFILERECORD     0x15
-#define MODBUS_FUNC_MASKWRITEREG        0x16
-#define MODBUS_FUNC_READWRITEMULTREGS   0x17
-#define MODBUS_FUNC_READFIFOQUEUE       0x18
-#define MODBUS_FUNC_ENCAPINTTRANS       0x2b
-#define MODBUS_FUNC_MASK                0x7f
-#define MODBUS_FUNC_ERRORMASK           0x80
-
-/* Modbus Diagnostic functions: Subfunction Code. */
-#define MODBUS_SUBFUNC_QUERY_DATA           0x00
-#define MODBUS_SUBFUNC_RESTART_COM          0x01
-#define MODBUS_SUBFUNC_DIAG_REGS            0x02
-#define MODBUS_SUBFUNC_CHANGE_DELIMITER     0x03
-#define MODBUS_SUBFUNC_LISTEN_MODE          0x04
-#define MODBUS_SUBFUNC_CLEAR_REGS           0x0a
-#define MODBUS_SUBFUNC_BUS_MSG_COUNT        0x0b
-#define MODBUS_SUBFUNC_COM_ERR_COUNT        0x0c
-#define MODBUS_SUBFUNC_EXCEPT_ERR_COUNT     0x0d
-#define MODBUS_SUBFUNC_SERVER_MSG_COUNT     0x0e
-#define MODBUS_SUBFUNC_SERVER_NO_RSP_COUNT  0x0f
-#define MODBUS_SUBFUNC_SERVER_NAK_COUNT     0x10
-#define MODBUS_SUBFUNC_SERVER_BUSY_COUNT    0x11
-#define MODBUS_SUBFUNC_SERVER_CHAR_COUNT    0x12
-#define MODBUS_SUBFUNC_CLEAR_COUNT          0x14
-
-/* Modbus Encapsulated Interface Transport function: MEI type. */
-#define MODBUS_MEI_ENCAPINTTRANS_CAN   0x0d
-#define MODBUS_MEI_ENCAPINTTRANS_READ  0x0e
-
-/* Modbus Exception Codes. */
-#define MODBUS_ERROR_CODE_ILLEGAL_FUNCTION      0x01
-#define MODBUS_ERROR_CODE_ILLEGAL_DATA_ADDRESS  0x02
-#define MODBUS_ERROR_CODE_ILLEGAL_DATA_VALUE    0x03
-#define MODBUS_ERROR_CODE_SERVER_DEVICE_FAILURE 0x04
-#define MODBUS_ERROR_CODE_MEMORY_PARITY_ERROR   0x08
-
-/* Modbus Application Protocol (MBAP) header. */
-struct ModbusHeader_ {
-    uint16_t     transactionId;
-    uint16_t     protocolId;
-    uint16_t     length;
-    uint8_t      unitId;
-}  __attribute__((__packed__));
-typedef struct ModbusHeader_ ModbusHeader;
-
-/* Modbus Read/Write function and Access Types. */
-#define MODBUS_TYP_WRITE_SINGLE         (MODBUS_TYP_WRITE | MODBUS_TYP_SINGLE)
-#define MODBUS_TYP_WRITE_MULTIPLE       (MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE)
-#define MODBUS_TYP_READ_WRITE_MULTIPLE  (MODBUS_TYP_READ | MODBUS_TYP_WRITE | MODBUS_TYP_MULTIPLE)
-
-/* Macro to convert quantity value (in bit) into count value (in word): count = Ceil(quantity/8) */
-#define CEIL(quantity) (((quantity) + 7)>>3)
-
-/* Modbus Default unreplied Modbus requests are considered a flood */
-#define MODBUS_CONFIG_DEFAULT_REQUEST_FLOOD 500
-
 /* Modbus default stream reassembly depth */
 #define MODBUS_CONFIG_DEFAULT_STREAM_DEPTH 0
```

Comment:
Only for unit tests, right ?

Reply:
Yes, so we could probably move this inside the `UNITTESTS` block.